python/triton/compiler/compiler.py

```diff
@@ -164,8 +164,11 @@ def optimize_ttgir(mod, num_stages, num_warps, num_ctas, target, cluster_info, e
     pm.add_tritongpu_remove_layout_conversions_pass()
     pm.add_tritongpu_decompose_conversions_pass()
     pm.add_tritongpu_ws_fixup_missing_attrs_pass()
-    if num_stages != 0:
+    if is_hip() and num_stages != 0:
+        pm.add_tritonamdgpu_reorder_instructions_pass()
+    else:
```

Comment:
This is the issue with gemm tuning.
When num_stages = 0, we enable stream-pipeline pass, which does not work with the old instruction-reorder pass so it disables it.
But now when num_stages = 0, this instruction-reorder pass is enabled.

Reply:
Yes, you are right about this. I fixed it and enabled amd reorder pass even when num_stages == 0, since this new reorder pass does not decrease performance in that case.

Reply:
No, when num_stages=0, we only want stream-pipeline pass.

Reply:
As agreed offline, we want to eventually enable both amd reorder pass and stream-pipelining. For now, we don't do reordering together with stream-pipelining.